samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -110,7 +113,18 @@ private enum SupportedFileType
                     return this.BadRequest($"Unsupported file type: {fileType}");
             }
 
-            await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm);
+            var existingMemorySource = (await this._chatMemorySourceRepository.FindByNameAsync(formFile.FileName)).FirstOrDefault();
+            var newMemorySource = new MemorySource(
+                documentImportForm.ChatId.ToString(),
+                formFile.FileName,
+                documentImportForm.UserName,
+                SourceType.File,
+                existingMemorySource?.Id,
+                null);
+
+            await this._chatMemorySourceRepository.UpsertAsync(newMemorySource);
+
+            await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm, newMemorySource.Id);
```

Comment:
Consider saving the memory source to the sourceRepo only after the ParseDocument has passed/completed, incase the parsing fails.

Reply:
fair point. I had used this approach to avoid having loose memory in the kernel if the save to storage failed for some reason... I could add a fallback here that deletes the upserted document (as long as it didn't already exist) if the parsing fails

Reply:
added a fallback. can you take another look>